src/pyhf/modifiers/staterror.py

```diff
@@ -123,19 +116,18 @@ def finalize(self, pdfconfig):
             pdfconfig.param_set(mod).sigmas = default_backend.tolist(sigmas[sigmas > 0])
 
     def apply(self, pars):
-        tensorlib, _ = get_backend()
-        if self.factor_access_indices is None:
+        if not self.parameters_helper.index_selection:
             return
-        select_from = tensorlib.concatenate([pars, self.default_value])
-        factor_row = tensorlib.gather(select_from, self.factor_access_indices)
-
-        results_staterr = tensorlib.einsum(
-            's,a,mb->msab',
-            tensorlib.astensor(self.sample_ones),
-            tensorlib.astensor(self.alpha_ones),
-            factor_row,
-        )
 
+        tensorlib, _ = get_backend()
+        if self.batch_size is None:
+            batched_pars = tensorlib.reshape(pars, (1,) + tensorlib.shape(pars))
+        else:
+            batched_pars = pars
+
+        flat_pars = tensorlib.reshape(batched_pars, (-1,))
+        statfactors = tensorlib.gather(flat_pars, self.access_field)
+        results_staterr = tensorlib.einsum('yab,s->ysab', statfactors, self.sample_ones)
```

Comment:
`y`?

Reply:
`yab`

* `y`: counter for instance of staterror
* `a`: batch direction
* `b`: global bin counter

Reply:
done